static/src/js/components/OrderOptionForm/__tests__/OrderOptionForm.test.jsx

```diff
@@ -255,6 +256,55 @@ describe('OrderOptionForm', () => {
   })
 
   describe('when getting and updating Order Option', () => {
+    describe('when getting a order option where the incoming data has errors', () => {
+      test('should show the error', async () => {
+        const navigateSpy = vi.fn()
+        vi.spyOn(router, 'useNavigate').mockImplementation(() => navigateSpy)
+
+        setup(
+          {
+            pageUrl: '/order-options/OO1000000-MMT_2/edit',
+            mocks: [{
+              request: {
+                query: GET_ORDER_OPTION,
+                variables: { params: { conceptId: 'OO1000000-MMT_2' } }
+              },
+              result: {
+                data: {
+                  orderOption: {
+                    associationDetails: {},
+                    conceptId: 'OO1000000-MMT_2',
+                    collections: {
+                      count: 0,
+                      items: []
+                    },
+                    deprecated: null,
+                    description: 'Test Description',
+                    form: 'Test Form',
+                    name: 'Test Name',
+                    nativeId: 'dce1859e-774c-4561-9451-fc9d77906015',
+                    pageTitle: 'Test Name',
+                    providerId: 'MMT_2',
+                    revisionId: '1',
+                    revisionDate: '2024-04-23T15:03:34.399Z',
+                    scope: 'PROVIDER',
+                    sortKey: '1234567',
+                    __typename: 'OrderOption'
+                  }
+                }
+              }
+            }]
+          }
+        )
+
+        await waitFor(async () => {
+          await screen.findByRole('textbox', { name: 'Name' })
```

Comment:
Why do we need the extra awaits? Seems to still work with just expect(screen.findByRole('textbox', { name: 'Name' }))

Reply:
The form data would not load without this await, Deep and I both looked at this, only way we could get this to work.